runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10258,41 +10253,39 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
          iCursor = generateRXInstruction(cg, TR::InstOpCode::getStoreOpCode(), node, sizeReg,
                       generateS390MemoryReference(metaReg, offsetof(J9VMThread, heapAlloc), cg), iCursor);
       TR::LabelSymbol * fillerRemLabel = generateLabelSymbol(cg);
-      TR::LabelSymbol * doneLabel = generateLabelSymbol(cg);
 
       TR::LabelSymbol * fillerLoopLabel = generateLabelSymbol(cg);
 
       // do this clear, if disableBatchClear is on
-      if (disableBatchClear && disableInitClear==NULL) //&& (node->getOpCodeValue() == TR::anewarray) && (node->getFirstChild()->getInt()>0) && (node->getFirstChild()->getInt()<6) )
+      if (clearMemory)
          {
          iCursor = generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, addressReg, resReg, iCursor);
-         // Dont overwrite the class
-         //
-         iCursor = generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, shiftReg, lengthReg, iCursor);
-         iCursor = generateRSInstruction(cg, TR::InstOpCode::SRA, node, shiftReg, 8);
 
+         if (TR::comp()->target().is64Bit())
+            {
+            iCursor = generateRSInstruction(cg, TR::InstOpCode::SRLG, node, shiftReg, lengthReg, 8, iCursor);
+            }
+         else
+            {
+            iCursor = generateRRInstruction(cg, TR::InstOpCode::LR, node, shiftReg, lengthReg, iCursor);
+            iCursor = generateRSInstruction(cg, TR::InstOpCode::SRL, node, shiftReg, shiftReg, 8, iCursor);
```

Comment:
SRL does not have the source operand - Can you confirm if this is correct ?  Also I think on z196 and newer we can use SRLK here that would save one instruction

Reply:
Fixed the issue and used SRLK on new version and 32bit config!